### DISCUSSION TITLE
VS2022 Copilot chat popup - Can't copy text

### DISCUSSION THREAD

**Discussion by trentsteel84** [community/community#66002] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

In the new copilot chat popup window that appears (not the tabbed one), it won't let me copy text from that window, whether i try to right click and copy or using Ctrl + C.
I'm using VS 17.7.3 and Copilot Chat 0.1.1305.23274

**Comment by Akash1134** [comment]
Hi @trentsteel84 ,

Could you please share an example illustrating the issue you're referring to? 

Alternatively, , please file a VS feedback ticket.

- Launch a developer prompt using any of the options mentioned here - https://learn.microsoft.com/en-us/visualstudio/ide/reference/command-prompt-powershell?view=vs-2022
- Report a problem from VS2022: https://aka.ms/vs/report-a-problem
- Run the command devenv /log (this will launch Visual Studio with verbose logging)
- Use the report a problem function to create a ticket
- Copy all the contents of the GitHub Copilot Output Window and add them to the ticket filed. 

Thanks!

**Reply by trentsteel84** [reply]
An example would be, if i right click on some code and click on "Ask Copilot" and then type in "Explain this method", in the popup i can highlight text, but i can't copy it. See the image below:

![image](https://github.com/community/community/assets/28557530/3fbfdb7d-c905-484e-97d8-5c6e65cfb5ef)

**Reply by rheapatel23** [reply]
Hi @trentsteel84, @Akash1134, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Reply by trentsteel84** [reply]
@rheapatel23 I can confirm that using CTRL + C on the highlighted text now works for me :)
However, if you right click on highlighted text, it seems to ignore the copilot window, because the copy option in the right click context window isn't available, so it seems like the right click is instead acting on the code behind the copilot window. So that could maybe be a further future improvement

**Comment by JochemPalmsens** [comment]
I'm having the same issue. When CoPilot gives an answer, it's not possible to copy the text.
To put it mildly: ChatGPT _does_ offer that. So Copilot should (at least) offer this too (since we're paying $20 per user for it).

Microsoft's "Report a problem" is useless. I tried a number of times to report issues. But they're either ignored, given low priority or they make you do the debugging.

**Comment by petersaighlee** [comment]
@trentsteel84, although I'm running into the same issue, I've found (sort of) a workaround. In the Copilot Chat window, you can ask in your prompt to respond inside a code block (even if it's not a response containing code). This doesn't always work, but when Copilot puts it in a code block, as requested, you can copy and paste the contents of the code block. I've tested it in the seperate (so not the inline-chat-window) chat window, and it works (but Copilot doesn't always put the response in a code block as requested).

Maybe this will help you for now.

**Reply by rheapatel23** [reply]
Hi @petersaighlee, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Reply by mikebollandajw** [reply]
i cant paste either :(

**Comment by rasidr** [comment]
Same issue with Copilot Chat version 0.1.1668.35657 and VS2022 17.7.5

**Reply by GenuineFlollie** [reply]
Please fix this, a lot of questions are not usefull if you can't copy the code. it's so great we where told during Techorama.

**Reply by GavinMannion** [reply]
Same issue on my side with version 1.133.0.0 and Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.7.6

Very frustrating

**Reply by rheapatel23** [reply]
Hi @rasidr, @GenuineFlollie, @GavinMannion, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Comment by datvm** [comment]
Same issue, may have happened before since I haven't used Copilot Chat for a while but it gives a good answer but there is no button for copying or putting it into the file. That's very frustrating. Just updated to VS 17.8 for .NET 8.

**Reply by rheapatel23** [reply]
Hi @datvm, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Comment by ayim** [comment]
CC: @rheapatel

**Comment by erikjsun** [comment]
Same problem here!

**Reply by rheapatel23** [reply]
Hi @erikjsun, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Comment by hjdrent** [comment]
This option needs to be added. I hope the developers will give some priority

**Reply by rheapatel23** [reply]
Hi @hjdrent, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Comment by GokhanAtilgan** [comment]
Same problem here! I want to share or save for next time.

**Reply by rheapatel23** [reply]
Hi @GokhanAtilgan, please download the latest version, this issue has been resolved :)

Thanks,
Rhea - Product Manager for GitHub Copilot Visual Studio

**Comment by ImMamey** [comment]
Same issue

**Reply by ayim** [reply]
@rheapatel23

**Reply by rheapatel23** [reply]
Hi @ImMamey, please download the latest version, this issue has been resolved :) 

Thanks, 
Rhea - Product Manager for GitHub Copilot Visual Studio

**Comment by RobertTempest** [comment]
Its not resolved. copy/paste and backspace don't work for me.

**Reply by ayim** [reply]
@RobertTempest would you mind filing a new bug using the Visual Studio "Report a Problem" feature?
That way, we can have the most accurate diagnosis to your specific case.
https://learn.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022

--
Aaron Yim
Product Manager for Visual Studio
aayim@microsoft.com

**Comment by sheng-di** [comment]
2024/4/27, I still have no way to copy the entire anwser as markdown. BUT in vscode I could.

**Comment by AyalMoran** [comment]
same problem. cant copy and Accept Solution button does nothing but close the Copilot Completions window.

Visual Studio Ver. 17.10.16,  GitHub Copilot  0.2.894.62884. System is updated and restarted.

**Comment by dengyakui** [comment]
Same issue, coule you just add a copy button like ChatGpt for users

**Comment by welcomehyunseo** [comment]
![image](https://github.com/user-attachments/assets/a74ef073-96c6-460f-b61b-81a9fd552af9)


Same issue.

It is not working to copy text by shortcuts or button clicks.

**Reply by adsouza-questrade** [reply]
Issue persists

**Reply by FreeGoStudio** [reply]
Issue persists

**Comment by mikebollandajw** [comment]
Proves nobody at MS using it if its just the customers having the problems, cant believe we still have this problem in 2026

**Reply by sheng-di** [reply]
Maybe it's just a KPI related project.
